### PR TITLE
Add Custom Geocoder

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
Adds a Pelias stitcher lambda function

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Node files
+node_modules
+.serverless
+.build
+dist
+
+# MacOS files
+.DS_Store
```
Comment (evansiroky) [inline .gitignore:8]: One of these files got checked into the `__tests__` folder.

File: serverless.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  secrets: ${file(env.yml)}
+functions:
+  autocomplete:
+    handler: handler.autocomplete
+    events:
+      - http:
+          method: get
+          path: autocomplete
+  search:
+    handler: handler.search
```
Comment (evansiroky) [inline serverless.yml:24]: Sorry for not noticing this earlier, but would we also need a handler.reverse in here?

**Review by evansiroky — CHANGES_REQUESTED** [review]
See 2 comments and also https://github.com/ibi-group/pelias-stitch/pull/2

File: .gitignore (hunk 1 above)
Comment (miles-grant-ibigroup) [reply .gitignore:8]: Thank you for catching that! Removed in a117ddfb6e2ecedfc19edefda7a76b42375cd147

File: serverless.yml (hunk 2 above)
Comment (miles-grant-ibigroup) [reply serverless.yml:24]: Absolutely! Added in 37ca4057a79f30604cbc1125127b10eabe06d9f5

**Comment by miles-grant-ibigroup** [comment]
Thank you for catching this now and not when we try to integrate this! Have also added some default behavior for if one of the geocodes doesn't respond properly (which happened to me as I was setting this up)

File: utils.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+  service: string,
+  query: string
+): Promise<FeatureCollection> => {
+  try {
+    const response = await fetch(`${baseUrl}/${service}?${query}`, {})
+    return await response.json()
+  } catch {
+    console.warn(`${baseUrl} failed to return response`)
+    return { features: [], type: 'FeatureCollection' }
+  }
```
Comment (evansiroky) [inline utils.ts:44]: This is going to end up swallowing errors that we may want to know about. Maybe we should add some Bugsnag reporting here.

**Review by evansiroky — APPROVED** [review]
Just one non-blocking comment.

File: utils.ts (hunk 3 above)
Comment (miles-grant-ibigroup) [reply utils.ts]: Added in 79923da2c255284e580df7907b9ec45b107762af

**Review by landonreed — APPROVED** [review]

**Pull request merged**